### PR TITLE
Various build updates

### PR DESCRIPTION
**Pull request by jamezp**
Use the JBoss Public Repository instead of the opensaml repository. The JBoss Public Repository will proxy the opensaml repository.

Remove hard-coded WildFly development version. Enable PR and push tests on wildfly-build instead of building WildFly in the maven CI runs.

[JBWS-4414] Exclude transitive dependencies from the feature-pack and explicitly add the required dependencies. This avoids overriding dependencies when provisioning that were brought in from other feature packs.

https://issues.redhat.com/browse/JBWS-4414

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1104,7 +1140,7 @@
       <id>wildflydev</id>
       <properties>
         <jboss.home>${server.home}</jboss.home>
-        <jboss.version>${wildfly.dev.version}</jboss.version>
+        <jboss.version>${wildfly.version}</jboss.version>
```
Comment (jbliznak) [inline pom.xml:1143]: what is the purpose of this profile after this change? it is the same as the one above now, can we get rid of it then?

Comment (jamezp) [reply pom.xml:1143]: We actually could rid of the profile, I was just trying to keep the change simple. However, you're right that it should probably just go away.

Comment (jamezp) [reply pom.xml]: I removed the profile

File: .github/workflows/wildfly-build.yml [hunk 2]
```diff
@@ -37,7 +43,7 @@ jobs:
           cache: 'maven'
       - name: Build with Maven Java ${{ matrix.java }} on WildFly ${{needs.wildfly-build.outputs.wildfly-version}}
         run: |
-          mvn -s ./.m2-settings.xml -B -V -fae -Pwildflydev clean install
+          mvn -s ./.m2-settings.xml -B -V -fae -Pwildflydev clean install '-Dwildfly.version=${{needs.wildfly-build.outputs.wildfly-version}}'
```
Comment (jbliznak) [inline .github/workflows/wildfly-build.yml:46]: missed one `wildflydev`

File: pom.xml (hunk 1 above)
Comment (jbliznak) [reply pom.xml]: the profile in root pomx.ml was in pair with https://github.com/jbossws/jbossws-cxf/blob/main/modules/testsuite/pom.xml#L667-L667 - that one seems to be also duplicated so I would say we can remove too

File: .github/workflows/wildfly-build.yml (hunk 2 above)
Comment (jamezp) [reply .github/workflows/wildfly-build.yml:46]: There is still one profile left which I assume is needed https://github.com/jbossws/jbossws-cxf/blob/004e5a30e87b7c991f2e11fd087bc7bcfca9c660/modules/testsuite/pom.xml#L671-L755.

Comment (jamezp) [reply .github/workflows/wildfly-build.yml:46]: Never mind, I see what you said above now :)

File: pom.xml (hunk 1 above)
Comment (jamezp) [reply pom.xml]: Oh, good catch. I'll remove this one too and the profile in the wildfly-build.yml.

**Comment by jimma** [comment]
This looks good. Thanks for the fix @jamezp !

**Pull request merged**